lib/path.cpp

```diff
@@ -245,13 +251,101 @@ bool Path::isHeader(const std::string &path)
     return startsWith(extension, ".h");
 }
 
-Standards::Language Path::identify(const std::string &path, bool *header)
+static bool hasEmacsCppMarker(const char* path)
+{
+    // TODO: identify is called three times for each file
+    // Preprocessor::loadFiles() -> createDUI()
+    // Preprocessor::preprocess() -> createDUI()
+    // TokenList::createTokens() -> TokenList::determineCppC()
+#ifdef LOG_EMACS_MARKER
+    std::cout << path << '\n';
+#endif
+
+    FILE *fp = fopen(path, "rt");
```

Comment:
shouldn't we look at the raw tokens from simplecpp instead? similar to when we extract inline suppressions.

Reply:
The seems like a very unnecessary dependency to add just to make sure we get the whole first line. That would also limit the cases where this could be used as we have code where we just want to identify the file but do not. (there is still much code which needs test coverage, cleanups and fixes).

It is also only done if a header or a extension-less file is explicitly specified for analysis so it will never be used during a regular analysis (possibly limited to certain IDE integrations only).

All the stray `Path::identify()` should also be gotten rid off and there should be only a single call when the filename is first seen. I want to store this information in `FileWithDetails`.

Reply:
ok